src/Controller/UI/GRPMetadataExportController.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace App\Controller\UI;
+
+use App\Entity\Dataset;
+use DateTime;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Serializer\Encoder\EncoderInterface;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+use Symfony\Component\Serializer\SerializerInterface as SerializerInterface;
+
+/**
+ * GRP metadata export report generator.
+ */
+class GRPMetadataExportController extends ReportController
+{
+    /**
+     * Var to hold serializer.
+     */
+    private $serializer;
+
+    /**
+     * Var to hold normalizer.
+     */
+    private $normalizer;
+
+    /**
+     * Var to hold encoder, needed for csv creation.
+     */
+    private $encoder;
+
+    /**
+     * Class constructor
+     *
+     */
+    public function __construct(SerializerInterface $serializer, NormalizerInterface $normalizer, EncoderInterface $encoder)
+    {
+        $this->serializer = $serializer;
+        $this->normalizer = $normalizer;
+        $this->encoder = $encoder;
```

Comment:
If these are not used globally, inject them at the "action".

Reply:
It also seems normalizer and encoder are not needed for csv only.